### PR TITLE
MM-45019_End user, post-trial: Create team restricted modal shows incorrect title/description

### PR DESCRIPTION
**Pull request by julmondragon**
#### Summary
Updating the copy in restricted feature modal for end-users (Prev/Post trial):

##### Create Team
- Title: Multiple teams available in paid plans
- Message: Multiple teams allow for context-specific spaces that are more attuned to your teams' needs.

##### User groups
- Title: User groups available in paid plans
- Message: User groups are a way to organize users and apply actions to all users within that group.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45019

###### Prev trial - Create team
Screen Shot 2022-06-15 at 3 06 15 PM

###### Post trial - Create team
Screen Shot 2022-06-15 at 3 28 02 PM

###### Prev trial - User groups
Screen Shot 2022-06-15 at 3 22 26 PM

###### Post trial - User groups
Screen Shot 2022-06-15 at 3 29 11 PM

###### Prev/Post trial - Create team
Screen Shot 2022-06-15 at 3 20 09 PM

###### Prev/Post trial - User Groups
Screen Shot 2022-06-15 at 3 20 26 PM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: components/feature_restricted_modal/feature_restricted_modal.tsx [hunk 1]
```diff
@@ -56,26 +61,43 @@ const FeatureRestrictedModal = ({
...
                 <p className='FeatureRestrictedModal__description'>
-                    {hasPriorTrial ? modalMessageAfterTrial : modalMessage}
+                    {getMessage()}
                 </p>
                 {showStartTrial && (
                     <p className='FeatureRestrictedModal__terms'>
                         <FormattedMessage
                             id='feature_restricted_modal.agreement'
-                            defaultMessage='By selecting <highlight>Try free for 30 days</highlight>, I agree to the <linkEvaluation>Mattermost Software Evaluation Agreement</linkEvaluation>, <linkPrivacy>Privacy Policy</linkPrivacy>, and receiving product emails.'
+                            defaultMessage='By selecting <highlight>Try free for {trialLength} days</highlight>, I agree to the <linkEvaluation>Mattermost Software Evaluation Agreement</linkEvaluation>, <linkPrivacy>Privacy Policy</linkPrivacy>, and receiving product emails.'
```
Comment (neallred) [inline components/feature_restricted_modal/feature_restricted_modal.tsx:98]: :+1:

**Review by neallred — APPROVED** [review]
LGTM, thanks @julmondragon !

**Review by pvev — APPROVED** [review]
LGTM! nice work @julmondragon :shipit:

**Review by stevemudie — APPROVED** [review]
Looks good for admins and users. Nice work @julmondragon!

**Pull request merged**